drv/i2c-devices/src/mwocp68.rs

```diff
@@ -27,15 +31,64 @@ pub struct Mwocp68 {
     mode: Cell<Option<pmbus::VOutModeCommandData>>,
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[repr(u8)]
+enum BootLoaderCommand {
+    ClearStatus = 0x00,
+    RestartProgramming = 0x01,
+    BootPrimary = 0x12,
+    BootSecondary = 0x02,
+    BootPSUFirmware = 0x03,
+}
+
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
 pub enum Error {
-    BadRead { cmd: u8, code: ResponseCode },
-    BadWrite { cmd: u8, code: ResponseCode },
-    BadData { cmd: u8 },
-    BadValidation { cmd: u8, code: ResponseCode },
-    InvalidData { err: pmbus::Error },
+    BadRead {
+        cmd: u8,
+        code: ResponseCode,
+    },
+    BadWrite {
+        cmd: u8,
+        code: ResponseCode,
+    },
+    BadData {
+        cmd: u8,
+    },
+    BadValidation {
+        cmd: u8,
+        code: ResponseCode,
+    },
+    InvalidData {
+        err: pmbus::Error,
+    },
+    BadFirmwareRevRead {
+        code: ResponseCode,
+    },
+    BadFirmwareRev {
+        index: u8,
+    },
+    BadFirmwareRevLength,
+    UpdateInBootLoader,
+    UpdateNotInBootLoader,
+    UpdateAlreadySuccessful,
+    BadBootLoaderStatus {
+        data: u8,
+    },
+    BadBootLoaderCommand {
+        cmd: BootLoaderCommand,
+        code: ResponseCode,
+    },
+    ChecksumNotSuccessful,
```

Comment:
here we've added a bunch of error variants that are only ever returned by specific functions, but are part of the error enum that's returned by _all_ functions in this module, forcing callers of those functions to handle the new update-specific variants. i'm not sure how important this is in practice, since presently, I think `sensor-polling` is the only caller of other methods on the `Mwocp68` type, and it handles any unexpected variant by just recording a `DeviceError` (rather than e.g. panicking)...so, it's probably fine.  but i figured it was worth mentioning.

Reply:
Yeah, I think it's okay -- and I really want to have the additional context that we log in the update task.